AutomatedTesting/Gem/PythonTests/Atom/atom_utils/material_canvas_utils.py

```diff
@@ -10,101 +10,101 @@
-    """
-    Given a slot_name string, return a GraphModelSlotId object representing a node slot.
-    :param slot_name: String representing the name of the slot to target on the node.
-    :return: An GraphModelSlotId object.
-    """
-    return graph.GraphModelSlotId(slot_name)
-
-
-def add_connection_by_slot_id(
-        graph_id: math.Uuid,
-        source_node: object, source_slot: object,
-        target_node: object, target_slot: object) -> object:
-    """
-    Adds a new connection between a source node slot and a target node slot.
-    :param graph_id: int representing the ID value of a given graph AZStd::shared_ptr<Graph> object.
-    :param source_node: A proxy AZStd::shared_ptr<Node> object for the source node to start the connection from.
-    :param source_slot: A proxy GraphModelSlotId object for the slot on the source node to use for the connection.
-    :param target_node: A proxy AZStd::shared_ptr<Node> object for the target node to end the connection to.
-    :param target_slot: A proxy GraphModelSlotId object for the slot on the target node to use for the connection.
-    :return: azlmbr.object.PythonProxyObject representing a C++ AZStd::shared_ptr<Connection> object.
-    """
-    return graph.GraphControllerRequestBus(
-        bus.Event, GraphControllerRequestBusEvents.ADD_CONNECTION_BY_SLOT_ID, graph_id,
-        source_node, source_slot, target_node, target_slot)
-
-
-def are_slots_connected(
-        graph_id: math.Uuid,
-        source_node: object, source_slot: object,
-        target_node: object, target_slot: object) -> bool:
-    """
-    Determines if 2 nodes have a connection formed between them and returns a boolean for success/failure.
-    :param graph_id: int representing the ID value of a given graph AZStd::shared_ptr<Graph> object.
-    :param source_node: An AZStd::shared_ptr<Node> object representing the source node for the connection.
-    :param source_slot: An GraphModelSlotId object representing the slot on the source node the connection uses.
-    :param target_node: An AZStd::shared_ptr<Node> object representing the target node for the connection.
-    :param target_slot: An GraphModelSlotId object representing the slot on the target node the connection uses.
-    :return: bool representing success (True) or failure (False).
-    """
-    return graph.GraphControllerRequestBus(
-        bus.Event, GraphControllerRequestBusEvents.ARE_SLOTS_CONNECTED, graph_id,
-        source_node, source_slot, target_node, target_slot)
+class Graph(object):
+    """
+    Opens the specified materialgraph file when inside MaterialCanvas which appears as a Graph.
+    Opening the Graph is required since it also sets the document_id for use with other functions.
+    """
+
+    def __init__(self, material_graph_file_path: str):
+        self.material_graph_file_path = material_graph_file_path
+        self.document_id = atom_tools_utils.open_document(self.material_graph_file_path)
```

Comment:
do we need to expose document model CRUD on the Graph object or should the object model be centered on the material_graph_file_path as a docuemnt which contains a graph.
what if I want to create a new graph from scratch (new document)
can I Graph.save_as(document_name:str)
should I be able to call Graph.close() or Graph.delete()

Reply:
Out of scope, that would be a periodic test case and this re-factor isn't trying to add any new functionality, just re-factor the existing functionality to use the new classes.
Reason being is the extra testing and debugging time may impact completion.